Checkout/Api/Platform/OrderPayments.php

```diff
@@ -20,7 +20,33 @@ public static function update(
         $payload = json_decode($request->getRawBody());
         /** @var Mage_Sales_Model_Order $order */
         $order = Mage::getModel('sales/order')->load($payload->payment->payment->parent_id);
-        if (self::isDelayedCapture($order) || $order->hasInvoices()) {
+
+        if ($payload->payment->transaction->txn_type === 'refund') {
+            $invoiceCollection = $order->getInvoiceCollection();
+            $invoice = $invoiceCollection->getFirstItem();
+
+            try {
+                self::creditmemo($order, $invoice, $payload->payment->transaction->txn_id);
+            } catch (Exception $e) {
+                $error = new stdClass();
+                $error->message = $e->getMessage();
+                $error->code = 500;
+                $error->type = 'server.internal_error';
+                return Bold_Checkout_Rest::buildResponse($response, json_encode(['errors' => [$error]]));
```

Comment:
If we have bugsnag connected to this project, this seems like a good place to log the exception.

Reply:
Oh, this is the module. What about writing the exception to the logger, then?

Reply:
yep, added!